cli/test/popper_test.py

```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+import shutil
+import contextlib
+import subprocess
+from git import Repo
+
+
+class PopperTest:
+
+    def __init__(self):
+
+        self.test_repo_path = '/tmp/mypaper'
+        self.ci = False
+
+    def delete_dir(self, path):
+
+        try:
+            os.rmdir(path)
+        except OSError as e:
+            os.system("docker run --rm -v /tmp:/tmp alpine:3.8 rm -rf "+path)
+
+    def init_test_repo(self):
+
+        path = self.test_repo_path
+        delete_dir(path)
+        os.mkdir(path)
+        file = open(path+'/README.md', 'w+')
+        file.write("foo")
+
+    def check_output(self, command, actions):
```

Comment:
the `click.testing.Result` class provides an `output` property containing the output of the command. So no need to use `subprocess` for this

Reply:
Actually, I forgot to mention that the output coming is only containing newlines. I think this is because the print is being done through internal functions and output is not being returned hence to pipe out the output I used subprocess.

Reply:
oh, you're right, everything that gets printed to stdout is via the logging framework. So we need to use the unit testing facilities for loggers as explained [here](https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.TestCase.assertLogs)